### DISCUSSION TITLE
Dotnet aspire for on prem/hybrid deployments

### DISCUSSION THREAD

**Discussion by thatstatsguy** [microsoft/aspire#6690] [General]
Hi there! 

Just started dipping my toes into the waters of .net aspire. I have a question related to the intended usage of dotnet aspire for on prem/hybrid deployments.

Having read the intro 
```
.NET Aspire is an opinionated, cloud ready stack for building observable, production ready, distributed applications.
```
What I get is that it's cloud ready, but doesn't explicitly exclude on prem/hybrid. Couple questions I have:
- Are there any considerations I should keep in mind when exploring this? 
- Were certain parts of aspire that were only intended for on cloud? I couldn't see 

Many thanks for any and all info!

**Comment by davidfowl** [comment]
> Are there any considerations I should keep in mind when exploring this?

The deployment tooling doesn't exist, but you can continue deploying like normal. In that case it's up to you to provide the required configuration for the application to run.

> Were certain parts of aspire that were only intended for on cloud? I couldn't see

The core of aspire has no ties to the cloud, but there are azure and aws specific resources, that only makes sense to use if you are going to a cloud. The only explicit other ties to cloud are the deployment tooling we (Microsoft) built out of the box is for azure container apps.
